src/main/java/org/jabref/gui/externalfiles/AutoSetFileLinksUtil.java

```diff
@@ -127,16 +139,60 @@ public List<LinkedFile> findAssociatedNotLinkedFiles(BibEntry entry) throws IOEx
 
             if (!fileAlreadyLinked) {
                 Optional<ExternalFileType> type = FileUtil.getFileExtension(foundFile)
-                                                            .map(extension -> ExternalFileTypes.getExternalFileTypeByExt(extension, filePreferences))
-                                                            .orElse(Optional.of(new UnknownExternalFileType("")));
+                                                          .map(extension -> ExternalFileTypes.getExternalFileTypeByExt(extension, filePreferences))
+                                                          .orElse(Optional.of(new UnknownExternalFileType("")));
 
                 String strType = type.isPresent() ? type.get().getName() : "";
                 Path relativeFilePath = FileUtil.relativize(foundFile, directories);
                 LinkedFile linkedFile = new LinkedFile("", relativeFilePath, strType);
                 linkedFiles.add(linkedFile);
             }
         }
-
         return linkedFiles;
     }
+
+    public RelinkedResults relinkingFiles(List<LinkedFile> listlinked) throws FileNotFoundException {
+        List<LinkedFile> changedFiles = new ArrayList<>();
+        List<String> exceptions = new ArrayList<>();
+
+        for (LinkedFile file : listlinked) {
+            Path path = Path.of(file.getLink());
+            if (!Files.exists(path)) {
+                Path filePath = Path.of(file.getLink());
+                // May need to put some limit since it can cause considerable performance problems.
+                String directoryPath = filePath.getParent().getParent().toString();
+
+                String fileNameString = filePath.getFileName().toString();
+
+                List<Path> fileLocations = searchFileInDirectoryAndSubdirectories(Path.of(directoryPath), fileNameString);
```

Comment:
Wrong start. Search in `databaseContext.getFileDirectories(filePreferences)` only.